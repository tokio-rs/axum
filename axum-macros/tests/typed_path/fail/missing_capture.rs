use axum_macros::TypedPath;
use serde::Deserialize;

#[derive(TypedPath, Deserialize)]
#[typed_path("/users")]
struct MyPath {
    id: u32,
}

fn main() {}
