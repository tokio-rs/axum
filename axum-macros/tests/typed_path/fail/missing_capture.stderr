error[E0027]: pattern does not mention field `id`
 --> tests/typed_path/fail/missing_capture.rs:5:14
  |
5 | #[typed_path("/users")]
  |              ^^^^^^^^ missing field `id`
  |
help: include the missing field in the pattern
  |
5 | #[typed_path("/users" { id })]
  |                       ++++++
help: if you don't care about this missing field, you can explicitly ignore it
  |
5 | #[typed_path("/users" { .. })]
  |                       ++++++
